File: 프로그래머스/0/181905. 문자열 뒤집기/문자열 뒤집기.js
function solution(my_string, s, e) {
    const before = my_string.slice(0, s);
    const middle = my_string.slice(s, e + 1).split('').reverse().join('');
    const after = my_string.slice(e + 1);

    return before + middle + after;
}

File: 프로그래머스/0/181905. 문자열 뒤집기/문자열 뒤집기.java
import java.util.*;

class Solution {
    public String solution(String my_string, int s, int e) {
        StringBuilder sb = new StringBuilder(my_string);
        
        e = Math.min(e, my_string.length() - 1);
        
        StringBuilder reversed = new StringBuilder(my_string.substring(s, e+1)).reverse();
        
        sb.replace(s, e+1, reversed.toString());
        
        return sb.toString();
    }
}